lib/src/dropdown/Dropdown.tsx

```diff
@@ -157,7 +157,7 @@ const DxcDropdown = ({
     return () => {
       window.removeEventListener("resize", handleMenuResize);
     };
-  }, []);
+  }, [label]);
```

Comment:
I don't think this is the right approach. Inside the array of dependencies should only go the `scope variables that are used by the side effect and change over time. From [React docs](https://reactjs.org/docs/hooks-reference.html#useeffect):

![image](https://user-images.githubusercontent.com/44321109/202435448-9d79b8b7-690c-4043-9843-deba2c185e1c.png)

The `label` variable is not been used by the `useEffect` so it cannot go inside the dependency array, even if its change indirectly affects the effect. Also, that was not the intended purpose of this side effect (it has to be executed only once to add the event to the listeners of the window, so when the viewport changes the list also changes).

Reply:
The problem here (if I understood Viti's issue correctly) is that first the dropdown renders and then changes its label with the user's username, so the list needs to adapt to that dynamic change. This is a different side effect which needs to be approached in a specific way. Maybe we should use the [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) for this purpose.